version: "3.7"

services:
  plex:
    container_name: plex
    image: cr.hotio.dev/hotio/plex
    networks:
      - proxy
    ports:
      - "32400:32400"
    environment:
      - PUID=1000
      - PGID=1000
      - UMASK=002
      - TZ=Canada/Eastern
      - PLEX_CLAIM
      - ADVERTISE_IP
      - ALLOWED_NETWORKS
      - PLEX_PASS=no
    volumes:
      - /home/samuel/docker/media/plex/config:/config
      - /home/samuel/docker/media/plex/transcode:/transcode
      - /mnt/nfs/data/media:/media
    device:
      - /dev/dri:/dev/dri

    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.plex-secure.rule=Host(`plex.local.getsamhome.com`)"
      - "traefik.http.routers.plex-secure.entrypoints=websecure"
      - "traefik.http.routers.plex-secure.tls=true"
      - "traefik.http.routers.plex-secure.service=plex"
      - "traefik.http.services.plex.loadbalancer.server.port=4533"
      - "traefik.docker.network=proxy"

networks:
  proxy:
    external: true
